machine/machine.go

```diff
@@ -51,24 +61,68 @@ func (r *Machine) Ping() error {
 	return nil
 }
 
+// StopAgent implements ControllerNode.StopAgent.
+func (r *Machine) StopAgent() error {
+	command := []string{"sudo", "systemctl", "stop", fmt.Sprintf("jujud-machine-%v", r.jujuID)}
+	var out string
+	var err error
+	if r.self {
+		out, err = runCommand(command...)
+	} else {
+		out, err = runRemoteCommand(r, command...)
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if out != "" {
+		return errors.Errorf("stop agent command should not have returned any output, but got %v", out)
+	}
+	return nil
+}
+
+// StartAgent implements ControllerNode.StartAgent.
+func (r *Machine) StartAgent() error {
+	command := []string{"sudo", "systemctl", "start", fmt.Sprintf("jujud-machine-%v", r.jujuID)}
+	var out string
+	var err error
+	if r.self {
+		out, err = runCommand(command...)
+	} else {
+		out, err = runRemoteCommand(r, command...)
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if out != "" {
+		return errors.Errorf("start agent command should not have returned any output, but got %v", out)
+	}
+	return nil
+}
+
 // runRemoteCommand takes in a command and runs it remotely using ssh.
 // All strings within the command must be individually passed-in.
 // For example,
 //     to run 'echo hi:D', pass in "echo", "hi:D",
 //     to stop juju-db, pass in "systemctl", "stop", "juju-db".
 var runRemoteCommand = func(r *Machine, commands ...string) (string, error) {
 	args := []string{
+		"sudo",
```

Comment:
It might simplify things if we require that juju-restore be run as root - that makes more sense to me than sudoing the commands it runs. I realise the remote commands still need to be sudoed since they ssh to ubuntu first, but that could be confined to runRemoteCommand.

Reply:
I am not sure that is the case all the time. I'd rather elevate privileges as needed.

For example, in my trials, I did not need root privileges to run mongorestore...

Reply:
I don't have a strong feeling about this - we can always change it later.